packages/amplify_authenticator/example/integration_test/sign_in_with_phone_test.dart

```diff
@@ -0,0 +1,180 @@
+    testWidgets('Sign in with unknown credentials', (tester) async {
+      final phone = generatePhone();
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      SignInPage signInPage = SignInPage(tester: tester);
+
+      // When I select my country code
+      await signInPage.selectCountryCode();
+
+      // And I type my "phone number" with status "UNKNOWN"
+      await signInPage.enterUsername(phone);
+
+      // And I type my password
+      await signInPage.enterPassword('UNKNOWN');
+
+      // And I click the "Sign in" button
+      await signInPage.submitSignIn();
+
+      // Then I see "User does not exist"
+      await signInPage.expectUserNotFound();
+    });
+
+    // Scenario: Sign in with unconfirmed credentials
+    testWidgets('Sign in with unconfirmed credentials', (tester) async {
+      final phone = generatePhone();
+      final password = generatePassword();
+      final email = generateEmail();
+
+      // Use the standard Amplify API to create the user in the Unconfirmed state
+      await Amplify.Auth.signUp(
+          username: phone,
+          password: password,
+          options: CognitoSignUpOptions(
+              userAttributes: {CognitoUserAttributeKey.email: email}));
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      SignInPage signInPage = SignInPage(tester: tester);
+      ConfirmSignUpPage confirmSignUpPage = ConfirmSignUpPage(tester: tester);
+
+      signInPage.expectUserNameIsPresent(usernameLabel: 'Phone Number');
+
+      // When I select my country code
+      await signInPage.selectCountryCode();
+
+      // When I type my "username" with status "unconfirmed"
+      await signInPage.enterUsername(phone);
+
+      // And I type my password
+      await signInPage.enterPassword(password);
+
+      // And I click the "Sign in" button
+      await signInPage.submitSignIn();
+
+      // Then I see "Confirmation Code"
+      confirmSignUpPage.expectConfirmationCodeIsPresent();
+
+      await deleteUser(phone);
+    });
+
+    // Scenario: Sign in with confirmed credentials then sign out
+    testWidgets('Sign in with confirmed credentials then sign out',
+        (tester) async {
+      final phone = generatePhone();
+      final password = generatePassword();
+      await adminCreateUser(phone, password,
+          autoConfirm: true, verifyAttributes: true);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      SignInPage signInPage = SignInPage(tester: tester);
+      signInPage.expectUserNameIsPresent(usernameLabel: 'Phone Number');
+
+      // When I type my "username" with status "UNKNOWN"
+      await signInPage.enterUsername(phone);
+
+      // And I type my password
+      await signInPage.enterPassword(password);
+
+      // And I click the "Sign in" button
+      await signInPage.submitSignIn();
+
+      await Future.delayed(const Duration(seconds: 4), () {});
```

Comment:
This is just waiting for the sign in screen to be visible, correct? Does `await tester.pumpAndSettle()` work here? If not, might be worth finding something that is more reliable without taking so long.

I would also move this to expectAuthenticated(). Generally the expectations are not async, but this is a scenario where making it async makes sense to me because there is an implied delay since sign out is not going to be visible right away - "The I (eventually) see Sign Out"

Reply:
Sorry - I already fixed the need for this and forgot to remove it.